unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -298,6 +401,38 @@ public static IntPtr get_exception_argument_null([NativeCallbackType("const char
         [NativeCallbackType("MonoType*")] IntPtr type)
         => type.TypeFromHandleIntPtr().ToNativeRepresentation();
 
+    // Keeping for reference for the move to C#
+    // Currently Type is not loaded
+    [NativeFunction(NativeFunctionOptions.DoNotGenerate)]
+    [return: NativeCallbackType("MonoClass*")]
+    public static IntPtr unity_class_get(
+        [NativeCallbackType("MonoImage*")] IntPtr image,
+        uint token)
+    {
+        Assembly assembly = image.AssemblyFromGCHandleIntPtr();
+        // Cache module acquisition to avoid unnecessary allocation of Module[]
+        var assemblyInfo = GetInfoForAssembly(assembly);
+        var typeHandle = assemblyInfo.module.ModuleHandle.GetRuntimeTypeHandleFromMetadataToken((int)token);
+        // Loads type if not already loaded
+        return Type.GetTypeFromHandle(typeHandle).TypeHandleIntPtr();
```

Comment:
Is this needed for correctness? Was returning the type handle value not enough?

Reply:
(same with `get_method` below)

Reply:
Yes, I view this as a TODO. Once all access is in managed, I expect initialization to be handled for us without tricks like this to force a load.

Reply:
@joncham I've added a task for this and updated the comment - https://jira.unity3d.com/browse/VM-2081